### PR TITLE
fix redis startup

### PR DESCRIPTION
**Pull request by fugufish**

### CODE REVIEW

File: lib/moleculer/transporters/redis.rb [hunk 1]
```diff
@@ -163,13 +165,21 @@ def subscribe(channel, &block)
             channel: channel,
             block:   block,
             config:  @config,
-          ).connect
+          )
+
+          @subscriptions.last.connect if @started
```
Comment (reecebisel) [inline lib/moleculer/transporters/redis.rb]: I think we could do a method that is a verbose check rather than being reliant on `@started`. 
```
def started?
  @started == true
end
```
This also helps if we change how we do the check around the check for started processes.

File: lib/moleculer/transporters/redis.rb [hunk 2]
```diff
@@ -186,13 +196,22 @@ def publish(packet)
...
         publisher.connect
+        subscriber.connect
       end
 
       def disconnect
         publisher.disconnect
         subscriber.disconnect
       end
 
+      def start
```
Comment (reecebisel) [inline lib/moleculer/transporters/redis.rb]: you could do an `alias :start, :connect` rather than the full method

File: spec/integration/transporters/redis_spec.rb [hunk 3]
```diff
@@ -1,46 +1,57 @@
...
+                                 } })
+    end
+
+    it "is capable of publishing packets and receiving them" do
+      received = false
+      subject.subscribe(packet.topic) do
+        received = true
+      end
+
+      sleep 0.5
```
Comment (reecebisel) [inline spec/integration/transporters/redis_spec.rb]: why is this here? Just to make sure we don't run into race conditions?

Comment (fugufish) [reply spec/integration/transporters/redis_spec.rb]: exactly that because of the way it starts up async we have to ensure that the subscription is active before we publish, there isn't  a way to detect that.
